### ISSUE TITLE
How to check if a user denied to receive push notifications on iOS?

### ISSUE THREAD

**Issue by tonholis** [CrossGeeks/FirebasePushNotificationPlugin#83]
I've implemented the following method in AppDelegate.cs and the variable *registered* is always TRUE even when I deny the app receiving Push Notifications on the initial permission dialog. 

```
public override void RegisteredForRemoteNotifications(UIApplication application, NSData deviceToken)
{
	var registered = UIApplication.SharedApplication.IsRegisteredForRemoteNotifications;

#if DEBUG
	FirebasePushNotificationManager.DidRegisterRemoteNotifications(deviceToken, FirebaseTokenType.Sandbox);
#else
	FirebasePushNotificationManager.DidRegisterRemoteNotifications(deviceToken, FirebaseTokenType.Production);
#endif

}
```

Also, I implemented the FailedToRegisterForRemoteNotifications method to check for errors and it is not fired when a user denies receiving push notifications. I'm not sure if this is the right place to check.

**Comment by rdelrosario** [comment]
I think you should get the OnNotificationError event fired when this happens.

**Comment by rdelrosario** [comment]
Released on the latest version: 2.3.0.

Now OnNotificationError event indicates the type of error on  Type property:
```cs
 public enum PushNotificationErrorType
    {
        Unknown,
        PermissionDenied,
        RegistrationFailed, //This when registration fails
        UnregistrationFailed
    }
```
